polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -212,6 +212,16 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
     const firstColumn = rows.map((row) => row.slice(0, 1));
     const firstHeading = headings.slice(0, 1);
     const firstTotal = totals?.slice(0, 1);
+    const tableHeaderRows = this.table.current?.children[0].childNodes;
+    const tableBodyRows = this.table.current?.children[1].childNodes;
+    const headerRowHeights: number[] = [];
+    const bodyRowHeights: number[] = [];
+    tableHeaderRows?.forEach((row: HTMLTableRowElement) => {
+      headerRowHeights.push(row.clientHeight);
+    });
+    tableBodyRows?.forEach((row: HTMLTableRowElement) => {
+      bodyRowHeights.push(row.clientHeight);
+    });
```

Comment:
In here you can simplify this by having a `getRowClienHeight` function, you can even put it outside of the component 

```suggestion
const tableHeaderRows = this.table.current?.children[0].childNodes;
const tableBodyRows = this.table.current?.children[1].childNodes;
const getRowClientHeight = (row: HTMLTableRowElement) => row.clientHeight;
const headerRowHeights: number[] = tableHeaderRows?.map(getRowClientHeight) || []; 
const bodyRowHeights: number[] = tableBodyRows?.map(getRowClientHeight) || [];
```

Reply:
Refactored as suggested but I did something a bit different since I can't use `.map()` on a `nodeList`.